lib/update.php

```diff
@@ -232,6 +235,24 @@ public function vimeography_check_for_missing_activation_keys() {
       }
     }
   }
+  
+  public function vimeography_check_for_expiring_keys(){
+    //Gather installed addon keys
+    $activation_keys = get_site_option('vimeography_activation_keys');
+    //Check addon expiration date
+    foreach($activation_keys as $activation_key){
+      $expiration_date = substr($activation_key->expires,0,10);
+      $today = date("Y-m-d");
+      $notification_date = date("Y-m-d",strtotime($expiration_date." -1 week"));
+      if($notification_date <= $today){
+        $this->expiring_addon = $activation_key;
```

Comment:
If multiple keys are expiring, this will overwrite the `expiring_addon` - perhaps an array would be better so that multiple expiring addons can be acted upon in the admin message display?